src/js/core/UpdateStream.js

```diff
@@ -0,0 +1,81 @@
+import { Observable } from "rxjs/Observable";
+import UserSettingsStore from "#SRC/js/stores/UserSettingsStore";
+import { request } from "@dcos/http-service";
+import Util from "#SRC/js/utils/Util";
+import { SAVED_STATE_KEY } from "#SRC/js/constants/UserSettings";
+
+const CHECK_DELAY = 24 * 60 * 60 * 1000; // once every 24 hours
+
+export const UpdateStreamType = Symbol("UpdateStreamType");
+
+function getVersionFromLocalStorage() {
+  const version = Util.findNestedPropertyInObject(
+    UserSettingsStore.getKey(SAVED_STATE_KEY),
+    "dcosUIVersion.version"
+  );
+
+  return version;
+}
+
+function setVersionInLocalStorage(version) {
+  const savedStates = UserSettingsStore.getKey(SAVED_STATE_KEY) || {};
+  savedStates.dcosUIVersion = { version };
+  UserSettingsStore.setKey(SAVED_STATE_KEY, savedStates);
+}
+
+function getContentType({ action, actionType, entity, version }) {
+  return `application/vnd.dcos.${entity}.${action}-${actionType}+json;charset=utf-8;version=${version}`;
+}
+
+function showNotification(newVersion) {
+  // TODO: show notification
+  console.log(
+    "A new version of DCOS UI is available. Install " +
+      newVersion +
+      ". Dismiss."
+  );
+}
+
+setVersionInLocalStorage("2.24.0"); // TODO: Check if a version exists
```

Comment:
This should contain the dismissed version numbers, no?

It might also be helpful to store the last time you checked for a new version, so you don't potentially trigger the notification every time the app loads.

Reply:
I am not sure what you mean with dismissed version numbers. I think it should contain the last version that is installed. The current "2.24.0" is just a stub, the jira ticket says setting it is in the scope of another ticket.

Reply:
Local storage will contain versions that the UI has notified a user about and the version the user has _dismissed_. We don't do manually `setVersionInLocalStorage`. Current version comes from `window.DCOS_UI_VERSION` as @brandonc pointed it the prev comment.

Reply:
@nLight Thank you, I understand now.